bridge/src/main/java/io/stargate/bridge/service/BridgeService.java

```diff
@@ -148,7 +148,7 @@ public void executeBatch(Batch batch, StreamObserver<Response> responseObserver)
   public void describeKeyspace(
       Schema.DescribeKeyspaceQuery request,
       StreamObserver<Schema.CqlKeyspaceDescribe> responseObserver) {
-    SchemaHandler.describeKeyspace(request, persistence, responseObserver);
+    executor.execute(() -> SchemaHandler.describeKeyspace(request, persistence, responseObserver));
```

Comment:
That looks good, but what about the `executeQueryWithSchema` method? Should suffer from same issues if the query is not executed?

Reply:
`executeQueryWithSchema` calls `executeQuery`, which creates a QueryHandler that uses the `executor` for `SchemaAgreementHelper` - is there something else that has to be scheduled on the executor? I thought we had said that queries themselves should remain on direct execution?

Reply:
But only in that `else` case, two other ifs are still processed in the same thread? execute query is not call always:

https://github.com/stargate/stargate/blob/d724cc5ed85f03c6f05b9606e192655495c6e9d2/bridge/src/main/java/io/stargate/bridge/service/BridgeService.java#L91-L138

Reply:
That said I am not sure what to actually do here, cause we don't want to offload query execution with two executor task.. Maybe we can leave as is, but that's not good imo as well..

Reply:
Well, if I took the body of those two `if` clauses and used the executor to execute them instead, that would work (and wouldn't spawn tasks within tasks or anything) - I will try that